MovieQuiz/Presentation/StatisticService.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  StatisticService.swift
+//  MovieQuiz
+//
+//  Created by Мария Солодова on 08.12.2022.
+//
+
+import Foundation
+
+protocol StatisticService {
+    var totalAccuracy: Double { get }
+    var gamesCount: Int { get }
+    var bestGame: GameRecord { get }
+    
+    func store(correct count: Int, total amount: Int)
+}
+
+final class StatisticServiceImplementation: StatisticService {
+    var totalAccuracy: Double {
+        get {
+            return Double(correct) / Double(total)
+        }
+    }
+    var gamesCount: Int {
+        get {
+            return userDefaults.integer(forKey: Keys.gamesCount.rawValue)
+        }
+        
+        set {
+            userDefaults.set(newValue, forKey: Keys.gamesCount.rawValue)
+        }
+    }
+
+
+    private enum Keys: String {
+        case correct, total, bestGame, gamesCount
+    }
+    
+    private let userDefaults = UserDefaults.standard
+    var bestGame: GameRecord {
+        get {
+            guard let data = userDefaults.data(forKey: Keys.bestGame.rawValue),
+            let record = try? JSONDecoder().decode(GameRecord.self, from: data) else {
+                return GameRecord(correct: 0, total: 0, date: Date())
+            }
+            
+            return record
+        }
+        
+        set {
+            guard let data = try? JSONEncoder().encode(newValue) else {
+                print("Невозможно сохранить результат")
+                return
+            }
+            
+            userDefaults.set(data, forKey: Keys.bestGame.rawValue)
+        }
+    }
+    
+    func store(correct count: Int, total amount: Int) {
+        let newRecord = GameRecord(correct: count, total: amount, date: Date())
+        
+        
+        if newRecord.isBetter(than: bestGame) {
+            bestGame = newRecord
+        }
+        
+        gamesCount += 1
+        correct += count
+        total += amount
+        
+    }
+    
+    
+    private var correct: Int {
+        get {
+            return userDefaults.integer(forKey: Keys.correct.rawValue)
+        }
+        set {
+            userDefaults.set(newValue, forKey: Keys.correct.rawValue)
+        }
+    }
+    
+    private var total: Int {
+        get {
+            return userDefaults.integer(forKey: Keys.total.rawValue)
+        }
+        set {
+            return userDefaults.set(newValue, forKey: Keys.total.rawValue)
+        }
+    }
+    
+    
+}
+
+
```

Comment:
между отдельными смысловыми блоками принято оставлять одну строку
```suggestion
    }
    
    private var correct: Int {
        get {
            return userDefaults.integer(forKey: Keys.correct.rawValue)
        }
        set {
            userDefaults.set(newValue, forKey: Keys.correct.rawValue)
        }
    }
    
    private var total: Int {
        get {
            return userDefaults.integer(forKey: Keys.total.rawValue)
        }
        set {
            return userDefaults.set(newValue, forKey: Keys.total.rawValue)
        }
    }
}
```